pkg/webhook/resources/setting/validator.go

```diff
@@ -179,19 +179,21 @@ func (v *settingValidator) validateBackupTarget(setting *v1beta1.Setting) error
 func (v *settingValidator) customizeTransport() error {
 	httpProxySetting, err := v.settingCache.Get(settings.HttpProxySettingName)
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to get HTTP proxy setting: %v", err)
 	}
 	var httpProxyConfig util.HTTPProxyConfig
-	if err := json.Unmarshal([]byte(httpProxySetting.Value), &httpProxyConfig); err != nil {
-		return err
+	if httpProxySetting.Value != "" {
+		if err := json.Unmarshal([]byte(httpProxySetting.Value), &httpProxyConfig); err != nil {
+			return fmt.Errorf("failed to parse HTTP proxy config: %v", err)
+		}
 	}
 	os.Setenv(util.HTTPProxyEnv, httpProxyConfig.HTTPProxy)
```

Comment:
Should those setting be put right before the final successful return? when any of following error happens, the ENV is set in a middle state.

Reply:
It is expected to use the configured proxy envs. We don't need to ensure it is set only when it succeeds